### ISSUE TITLE
Crashes in ASAssertUnlocked(__instanceLock__)

### ISSUE THREAD

**Issue by hashemp206** [TextureGroup/Texture#1094]
I'm using the latest commit from branch master, but `ASDisplayNode` crashes in `_didLoad` method

any idea?

> 'NSInternalInconsistencyException', reason: 'Expected __instanceLock__.locked() to be false.'

<img width="342" alt="screen shot 2018-09-01 at 8 37 05 am" src="https://user-images.githubusercontent.com/3206748/44942215-3b99dc80-adc2-11e8-9612-aba3c75e7650.png">

<img width="804" alt="screen shot 2018-09-01 at 8 35 55 am" src="https://user-images.githubusercontent.com/3206748/44942212-2755df80-adc2-11e8-8a06-6ac05f268206.png">

**Comment by maicki** [comment]
@hashemp206 I'm looking into something similar at the moment and at least know why the stack trace looks like. Can you please share the the output if you do `bt all` in the console if the assert is hit. Thanks!

**Comment by hashemp206** [comment]
```
* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 1.1
    frame #0: 0x0000000209fb5a08 libobjc.A.dylib`objc_exception_throw
    frame #1: 0x000000020acfc130 CoreFoundation`+[NSException raise:format:arguments:] + 104
    frame #2: 0x000000020b7e3560 Foundation`-[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 112
  * frame #3: 0x000000010420506c AsyncDisplayKit`::-[ASDisplayNode _didLoad](self=0x00000001442a4f40, _cmd="_didLoad") at ASDisplayNode.mm:552
    frame #4: 0x0000000104205c24 AsyncDisplayKit`::-[ASDisplayNode view](self=0x00000001442a4f40, _cmd="view") at ASDisplayNode.mm:644
    frame #5: 0x000000010419ddd4 AsyncDisplayKit`::-[ASCollectionNode view](self=0x00000001442a4f40, _cmd="view") at ASCollectionNode.mm:233
    frame #6: 0x000000010419e164 AsyncDisplayKit`::-[ASCollectionNode rangeController](self=0x00000001442a4f40, _cmd="rangeController") at ASCollectionNode.mm:291
    frame #7: 0x000000010419de3c AsyncDisplayKit`::-[ASCollectionNode clearContents](self=0x00000001442a4f40, _cmd="clearContents") at ASCollectionNode.mm:239
    frame #8: 0x000000010421e5b8 AsyncDisplayKit`::__44-[ASDisplayNode applyPendingInterfaceState:]_block_invoke(.block_descriptor=0x000000014573b980) at ASDisplayNode.mm:3152
    frame #9: 0x0000000105d3f908 libdispatch.dylib`_dispatch_call_block_and_release + 24
    frame #10: 0x0000000105d40eac libdispatch.dylib`_dispatch_client_callout + 16
    frame #11: 0x0000000105d4ea94 libdispatch.dylib`_dispatch_main_queue_callback_4CF + 1360
    frame #12: 0x000000020ad761a8 CoreFoundation`__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12
    frame #13: 0x000000020ad71070 CoreFoundation`__CFRunLoopRun + 1964
    frame #14: 0x000000020ad705a4 CoreFoundation`CFRunLoopRunSpecific + 436
    frame #15: 0x000000020cfde584 GraphicsServices`GSEventRunModal + 100
    frame #16: 0x0000000237337b04 UIKitCore`UIApplicationMain + 212
    frame #17: 0x00000001012710b0 Fidilio`main at AppDelegate.swift:13
    frame #18: 0x000000020a830dc8 libdyld.dylib`start + 4
  thread #4, name = 'gputools.smt_poll.0x141f5eba0'
    frame #0: 0x000000020a97d428 libsystem_kernel.dylib`__semwait_signal + 8
    frame #1: 0x000000020a8f25d0 libsystem_c.dylib`nanosleep + 212
    frame #2: 0x000000020a8f24a4 libsystem_c.dylib`usleep + 64
    frame #3: 0x0000000105e56ed0 GPUToolsCore`smt_poll_thread_entry(void*) + 136
    frame #4: 0x000000020aa012fc libsystem_pthread.dylib`_pthread_body + 128
    frame #5: 0x000000020aa0125c libsystem_pthread.dylib`_pthread_start + 48
    frame #6: 0x000000020aa04d08 libsystem_pthread.dylib`thread_start + 4
  thread #6, name = 'gputools.smt_poll.0x141e04ea0'
    frame #0: 0x000000020a97d428 libsystem_kernel.dylib`__semwait_signal + 8
    frame #1: 0x000000020a8f25d0 libsystem_c.dylib`nanosleep + 212
    frame #2: 0x000000020a8f24a4 libsystem_c.dylib`usleep + 64
    frame #3: 0x0000000105e56ed0 GPUToolsCore`smt_poll_thread_entry(void*) + 136
    frame #4: 0x000000020aa012fc libsystem_pthread.dylib`_pthread_body + 128
    frame #5: 0x000000020aa0125c libsystem_pthread.dylib`_pthread_start + 48
    frame #6: 0x000000020aa04d08 libsystem_pthread.dylib`thread_start + 4
  thread #8, name = 'com.apple.uikit.eventfetch-thread'
    frame #0: 0x000000020a971ed0 libsystem_kernel.dylib`mach_msg_trap + 8
    frame #1: 0x000000020a9713a8 libsystem_kernel.dylib`mach_msg + 72
    frame #2: 0x000000020ad75f9c CoreFoundation`__CFRunLoopServiceMachPort + 236
    frame #3: 0x000000020ad70e38 CoreFoundation`__CFRunLoopRun + 1396
    frame #4: 0x000000020ad705a4 CoreFoundation`CFRunLoopRunSpecific + 436
    frame #5: 0x000000020b75f824 Foundation`-[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 300
    frame #6: 0x000000020b75f6d0 Foundation`-[NSRunLoop(NSRunLoop) runUntilDate:] + 148
    frame #7: 0x00000002372950a0 UIKitCore`-[UIEventFetcher threadMain] + 136
    frame #8: 0x000000020b892438 Foundation`__NSThread__start__ + 1040
    frame #9: 0x000000020aa012fc libsystem_pthread.dylib`_pthread_body + 128
    frame #10: 0x000000020aa0125c libsystem_pthread.dylib`_pthread_start + 48
    frame #11: 0x000000020aa04d08 libsystem_pthread.dylib`thread_start + 4
  thread #9, queue = 'BKSAssertionClientMultiplexer'
    frame #0: 0x000000020a971ed0 libsystem_kernel.dylib`mach_msg_trap + 8
    frame #1: 0x000000020a9713a8 libsystem_kernel.dylib`mach_msg + 72
    frame #2: 0x0000000105d596a8 libdispatch.dylib`_dispatch_mach_send_and_wait_for_reply + 500
    frame #3: 0x0000000105d59b3c libdispatch.dylib`dispatch_mach_send_with_result_and_wait_for_reply + 56
    frame #4: 0x000000020aa39144 libxpc.dylib`xpc_connection_send_message_with_reply_sync + 204
    frame #5: 0x000000020d5f516c BaseBoard`-[BSXPCMessage sendSynchronouslyToConnection:error:] + 84
    frame #6: 0x000000020d5f4d6c BaseBoard`-[BSBaseXPCClient _sendMessage:withReplyHandler:waitForReply:waitDuration:] + 316
    frame #7: 0x000000020d6766bc AssertionServices`-[BKSAssertionClientMultiplexer sendEvent:withAssertionType:forEventType:responseHandler:] + 256
    frame #8: 0x000000020d67050c AssertionServices`-[BKSProcessAssertionClient sendCreateEvent:error:] + 116
    frame #9: 0x000000020d674004 AssertionServices`-[BKSAssertion _clientQueue_acquireAssertion] + 136
    frame #10: 0x000000020d673c30 AssertionServices`__45-[BKSAssertion _registerAssertionAndAcquire:]_block_invoke + 80
    frame #11: 0x0000000105d3f908 libdispatch.dylib`_dispatch_call_block_and_release + 24
    frame #12: 0x0000000105d40eac libdispatch.dylib`_dispatch_client_callout + 16
    frame #13: 0x0000000105d48ee0 libdispatch.dylib`_dispatch_lane_serial_drain + 720
    frame #14: 0x0000000105d49c00 libdispatch.dylib`_dispatch_lane_invoke + 516
    frame #15: 0x0000000105d53c18 libdispatch.dylib`_dispatch_workloop_worker_thread + 1220
    frame #16: 0x000000020aa020f0 libsystem_pthread.dylib`_pthread_wqthread + 312
    frame #17: 0x000000020aa04d00 libsystem_pthread.dylib`start_wqthread + 4
  thread #10
    frame #0: 0x000000020a97db9c libsystem_kernel.dylib`__workq_kernreturn + 8
    frame #1: 0x000000020aa021d4 libsystem_pthread.dylib`_pthread_wqthread + 540
    frame #2: 0x000000020aa04d00 libsystem_pthread.dylib`start_wqthread + 4
  thread #11
    frame #0: 0x000000020aa04cfc libsystem_pthread.dylib`start_wqthread
  thread #12, name = 'GAIThread', queue = 'SQLQueue 0x14368ca60 for googleanalytics-v3.sql'
    frame #0: 0x000000020a971e1c libsystem_kernel.dylib`_kernelrpc_mach_vm_map_trap + 8
    frame #1: 0x000000020a97b6b4 libsystem_kernel.dylib`mach_vm_map + 120
    frame #2: 0x000000020a9bffec libsystem_malloc.dylib`mvm_allocate_pages + 224
    frame #3: 0x000000020a9b9dd4 libsystem_malloc.dylib`large_malloc + 132
    frame #4: 0x000000020a9ba92c libsystem_malloc.dylib`szone_malloc_should_clear + 160
    frame #5: 0x000000020a9c0b0c libsystem_malloc.dylib`stack_logging_lite_memalign + 52
    frame #6: 0x000000020a9c4080 libsystem_malloc.dylib`malloc_zone_memalign + 228
    frame #7: 0x0000000209fcfdd0 libobjc.A.dylib`(anonymous namespace)::AutoreleasePoolPage::autoreleaseFullPage(objc_object*, (anonymous namespace)::AutoreleasePoolPage*) + 48
    frame #8: 0x000000020adf64d0 CoreFoundation`_CFAutoreleasePoolPush + 16
    frame #9: 0x000000020ad536d8 CoreFoundation`_CFXNotificationPost + 156
    frame #10: 0x000000020b75e030 Foundation`-[NSNotificationCenter postNotificationName:object:userInfo:] + 68
    frame #11: 0x000000020db58790 CoreData`_beginTransactionForSaveRequest + 120
    frame #12: 0x000000020db595c8 CoreData`_executeSaveChangesRequest + 248
    frame #13: 0x000000020d8f8790 CoreData`-[NSSQLSaveChangesRequestContext executeRequestCore:] + 24
    frame #14: 0x000000020d8f64c4 CoreData`-[NSSQLStoreRequestContext executeRequestUsingConnection:] + 252
    frame #15: 0x000000020da5a340 CoreData`__52-[NSSQLDefaultConnectionManager handleStoreRequest:]_block_invoke + 84
    frame #16: 0x0000000105d40eac libdispatch.dylib`_dispatch_client_callout + 16
    frame #17: 0x0000000105d4f848 libdispatch.dylib`_dispatch_lane_barrier_sync_invoke_and_complete + 124
    frame #18: 0x000000020d8f632c CoreData`-[NSSQLDefaultConnectionManager handleStoreRequest:] + 256
    frame #19: 0x000000020d8f61e8 CoreData`-[NSSQLCoreDispatchManager routeStoreRequest:] + 264
    frame #20: 0x000000020d8f5fa4 CoreData`-[NSSQLCore dispatchRequest:withRetries:] + 260
    frame #21: 0x000000020d8f8570 CoreData`-[NSSQLCore processSaveChanges:forContext:] + 204
    frame #22: 0x000000020d8f6f2c CoreData`-[NSSQLCore executeRequest:withContext:error:] + 660
    frame #23: 0x000000020da64938 CoreData`__65-[NSPersistentStoreCoordinator executeRequest:withContext:error:]_block_invoke + 1980
    frame #24: 0x000000020d8f5d94 CoreData`-[NSPersistentStoreCoordinator _routeHeavyweightBlock:] + 240
    frame #25: 0x000000020d8f6a24 CoreData`-[NSPersistentStoreCoordinator executeRequest:withContext:error:] + 860
    frame #26: 0x000000020d8f78a8 CoreData`-[NSManagedObjectContext save:] + 1568
    frame #27: 0x00000001017e0070 Fidilio`__21-[GAIDataStore save:]_block_invoke + 168
    frame #28: 0x00000001017ddb88 Fidilio`-[GAIDataStore performBlockAndWait:withError:] + 380
    frame #29: 0x00000001017dff88 Fidilio`-[GAIDataStore save:] + 92
    frame #30: 0x00000001017ed1d4 Fidilio`-[GAIBatchingDispatcher persist:] + 136
    frame #31: 0x00000001017ee7e8 Fidilio`-[GAIBatchingDispatcher queueDispatch:] + 700
    frame #32: 0x00000001017ee510 Fidilio`-[GAIBatchingDispatcher queueModel:] + 136
    frame #33: 0x000000020b892628 Foundation`__NSThreadPerformPerform + 336
    frame #34: 0x000000020ad765a4 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24
    frame #35: 0x000000020ad76524 CoreFoundation`__CFRunLoopDoSource0 + 88
    frame #36: 0x000000020ad75e08 CoreFoundation`__CFRunLoopDoSources0 + 176
    frame #37: 0x000000020ad70cd4 CoreFoundation`__CFRunLoopRun + 1040
    frame #38: 0x000000020ad705a4 CoreFoundation`CFRunLoopRunSpecific + 436
    frame #39: 0x000000020b75f824 Foundation`-[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 300
    frame #40: 0x000000020b79b214 Foundation`-[NSRunLoop(NSRunLoop) run] + 88
    frame #41: 0x00000001017dba4c Fidilio`+[GAI threadMain:] + 64
    frame #42: 0x000000020b892438 Foundation`__NSThread__start__ + 1040
    frame #43: 0x000000020aa012fc libsystem_pthread.dylib`_pthread_body + 128
    frame #44: 0x000000020aa0125c libsystem_pthread.dylib`_pthread_start + 48
    frame #45: 0x000000020aa04d08 libsystem_pthread.dylib`thread_start + 4
  thread #13, queue = 'com.google.fira.worker'
    frame #0: 0x000000020a97e430 libsystem_kernel.dylib`fsync + 8
    frame #1: 0x000000020b24e5b0 libsqlite3.dylib`___lldb_unnamed_symbol159$$libsqlite3.dylib + 268
    frame #2: 0x000000020b2590b8 libsqlite3.dylib`___lldb_unnamed_symbol193$$libsqlite3.dylib + 516
    frame #3: 0x000000020b24ded0 libsqlite3.dylib`___lldb_unnamed_symbol156$$libsqlite3.dylib + 1284
    frame #4: 0x000000020b2384b4 libsqlite3.dylib`___lldb_unnamed_symbol122$$libsqlite3.dylib + 164
    frame #5: 0x000000020b203a88 libsqlite3.dylib`___lldb_unnamed_symbol41$$libsqlite3.dylib + 2788
    frame #6: 0x000000020b230708 libsqlite3.dylib`___lldb_unnamed_symbol91$$libsqlite3.dylib + 59516
    frame #7: 0x000000020b220af4 libsqlite3.dylib`sqlite3_step + 432
    frame #8: 0x000000020b1e9380 libsqlite3.dylib`sqlite3_exec + 336
    frame #9: 0x00000001018533c4 Fidilio`-[FIRASqliteStore endTransaction] + 72
    frame #10: 0x0000000101853084 Fidilio`-[FIRASqliteStore performTransaction:] + 116
    frame #11: 0x00000001018715b8 Fidilio`-[FIRAMeasurement writeEventOnWorkerQueue:] + 1176
    frame #12: 0x0000000101870fd4 Fidilio`-[FIRAMeasurement handleEventOnWorkerQueue:] + 384
    frame #13: 0x00000001018840a4 Fidilio`__52-[FIRAScheduler scheduleOnWorkerQueueBlockID:block:]_block_invoke + 48
    frame #14: 0x0000000105d3f908 libdispatch.dylib`_dispatch_call_block_and_release + 24
    frame #15: 0x0000000105d40eac libdispatch.dylib`_dispatch_client_callout + 16
    frame #16: 0x0000000105d48ee0 libdispatch.dylib`_dispatch_lane_serial_drain + 720
    frame #17: 0x0000000105d49bc8 libdispatch.dylib`_dispatch_lane_invoke + 460
    frame #18: 0x0000000105d53c18 libdispatch.dylib`_dispatch_workloop_worker_thread + 1220
    frame #19: 0x000000020aa020f0 libsystem_pthread.dylib`_pthread_wqthread + 312
    frame #20: 0x000000020aa04d00 libsystem_pthread.dylib`start_wqthread + 4
  thread #16
    frame #0: 0x000000020a97db9c libsystem_kernel.dylib`__workq_kernreturn + 8
    frame #1: 0x000000020aa02114 libsystem_pthread.dylib`_pthread_wqthread + 348
    frame #2: 0x000000020aa04d00 libsystem_pthread.dylib`start_wqthread + 4
  thread #19, name = 'com.apple.NSURLConnectionLoader'
    frame #0: 0x000000020a971ed0 libsystem_kernel.dylib`mach_msg_trap + 8
    frame #1: 0x000000020a9713a8 libsystem_kernel.dylib`mach_msg + 72
    frame #2: 0x000000020ad75f9c CoreFoundation`__CFRunLoopServiceMachPort + 236
    frame #3: 0x000000020ad70e38 CoreFoundation`__CFRunLoopRun + 1396
    frame #4: 0x000000020ad705a4 CoreFoundation`CFRunLoopRunSpecific + 436
    frame #5: 0x000000020b38de54 CFNetwork`-[__CoreSchedulingSetRunnable runForever] + 212
    frame #6: 0x000000020b892438 Foundation`__NSThread__start__ + 1040
    frame #7: 0x000000020aa012fc libsystem_pthread.dylib`_pthread_body + 128
    frame #8: 0x000000020aa0125c libsystem_pthread.dylib`_pthread_start + 48
    frame #9: 0x000000020aa04d08 libsystem_pthread.dylib`thread_start + 4
  thread #20
    frame #0: 0x000000020aa04cfc libsystem_pthread.dylib`start_wqthread
  thread #23, name = 'com.google.Maps.LabelingBehavior'
    frame #0: 0x000000020a971ed0 libsystem_kernel.dylib`mach_msg_trap + 8
    frame #1: 0x000000020a9713a8 libsystem_kernel.dylib`mach_msg + 72
    frame #2: 0x000000020ad75f9c CoreFoundation`__CFRunLoopServiceMachPort + 236
    frame #3: 0x000000020ad70e38 CoreFoundation`__CFRunLoopRun + 1396
    frame #4: 0x000000020ad705a4 CoreFoundation`CFRunLoopRunSpecific + 436
    frame #5: 0x000000020b75f824 Foundation`-[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 300
    frame #6: 0x0000000101c39084 Fidilio`-[GMSx_GTMSimpleWorkerThread main] + 524
    frame #7: 0x000000020b892438 Foundation`__NSThread__start__ + 1040
    frame #8: 0x000000020aa012fc libsystem_pthread.dylib`_pthread_body + 128
    frame #9: 0x000000020aa0125c libsystem_pthread.dylib`_pthread_start + 48
    frame #10: 0x000000020aa04d08 libsystem_pthread.dylib`thread_start + 4
  thread #24, name = 'ASDeallocQueue'
    frame #0: 0x000000020a971ed0 libsystem_kernel.dylib`mach_msg_trap + 8
    frame #1: 0x000000020a9713a8 libsystem_kernel.dylib`mach_msg + 72
    frame #2: 0x000000020ad75f9c CoreFoundation`__CFRunLoopServiceMachPort + 236
    frame #3: 0x000000020ad70e38 CoreFoundation`__CFRunLoopRun + 1396
    frame #4: 0x000000020ad705a4 CoreFoundation`CFRunLoopRunSpecific + 436
    frame #5: 0x000000020ad71340 CoreFoundation`CFRunLoopRun + 80
    frame #6: 0x000000010429c0a0 AsyncDisplayKit`::-[ASDeallocQueueV1 threadMain](self=0x000000014532d840, _cmd="threadMain") at ASRunLoopQueue.mm:121
    frame #7: 0x000000020b892438 Foundation`__NSThread__start__ + 1040
    frame #8: 0x000000020aa012fc libsystem_pthread.dylib`_pthread_body + 128
    frame #9: 0x000000020aa0125c libsystem_pthread.dylib`_pthread_start + 48
    frame #10: 0x000000020aa04d08 libsystem_pthread.dylib`thread_start + 4
  thread #25
    frame #0: 0x0000000000000000
```

**Comment by maicki** [comment]
Should be fixed with #1103
